### ISSUE TITLE
texture demo issue

### ISSUE THREAD

**Issue by petergerten** [YadiraF/PRNet#30]
Hi,

thanks for the great project!
I have some trouble to get the texture demo working.

I get:

```
  File "demo_texture.py", line 101, in <module>
    texture_editing(prn, parser.parse_args())
  File "demo_texture.py", line 67, in texture_editing
    new_colors = prn.get_colors_from_texture(new_texture)
AttributeError: PRN instance has no attribute 'get_colors_from_texture'
```

**Comment by petergerten** [comment]
I checked the code and it seems 'get_colors_from_texture' is indeed missing. Maybe you did not commit an updated api.py?

**Comment by YadiraF** [comment]
Yes, sorry for forgetting to update this file. I've just fixed. 
Thanks for your reporting.

**Comment by petergerten** [comment]
@YadiraF thanks! Got if working now.
While the speed of the core system is impressive, I have a question regarding the texture demo:
It seems that the render_texture() util function is **very** slow (~3-4 seconds) on my system. Do you have any suggestion/idea on how to improve this? I really do not know much about this but it seems that the nested for loops might slow things down and potentially could be vectorized?
